ui/app/AppLayouts/Chat/views/CommunityColumnView.qml

```diff
@@ -284,19 +284,19 @@ Item {
 
             categoryPopupMenu: StatusMenu {
 
-                property var itemWithCategory
+                property var categoryItem
                 property string categoryId
 
                 openHandler: function () {
-                    let jsonObj = root.communitySectionModule.getItemPartOfCategoryAsJsonById(categoryId)
+                    let jsonObj = root.communitySectionModule.getCategoryItemAsJsonById(categoryId)
```

Comment:
I don't see the need for this imperative approach; you should have all the data coming from the model for each item.

Reply:
Done. It's much cleaner

Reply:
Great; still too much JS there but I like where this is going :)